File: FrondEndAngular/staffBoxe/src/app/common/purchase.ts
import { Address } from "./address";
import { Adherent } from "./adherent";
import { Order } from "./order";
import { OrderItem } from "./orderItem";
import { User } from "./user";

export class Purchase{
    user!: User;
    adherentCreate! : Adherent;
    shippingAddress! : Address;
    billingAddress !: Address;
    order!: Order;
    orderItems!: OrderItem[];

    constructor(){
   
    }
}

File: src/main/java/com/asso/staff/dto/Purchase.java
package com.asso.staff.dto;

import java.util.Set;

import com.asso.staff.entity.Address;
import com.asso.staff.entity.Order;
import com.asso.staff.entity.OrderItem;
import com.asso.staff.entity.User;

import lombok.Data;

@Data
public class Purchase {

	private User user;
    private AdherentCreateDTO adherentCreate;
	private Address shippingAddress;
	private Address billingAddress;
	private Order order;
	private Set<OrderItem> orderItems;
	
}
